public/app/features/dashboard-scene/scene/DashboardSceneUrlSync.ts

```diff
@@ -84,7 +86,29 @@ export class DashboardSceneUrlSync implements SceneObjectUrlSyncHandler {
       update.viewPanelScene = undefined;
     }
 
+    // Handle edit panel state
+    if (typeof values.editPanel === 'string') {
+      const panel = findVizPanelByKey(this._scene, values.editPanel);
+      if (!panel) {
+        return;
+      }
+
+      this._scene.onEnterEditMode();
+      update.editPanel = buildPanelEditScene(panel);
+    } else if (editPanel) {
+      update.editPanel = undefined;
+    }
+
     if (Object.keys(update).length > 0) {
+      if (update.hasOwnProperty('inspectPanelKey')) {
+        if (update.viewPanelScene === undefined) {
+          delete update.viewPanelScene;
+        }
+        if (update.editPanel === undefined) {
+          delete update.editPanel;
+        }
+      }
```

Comment:
not sure I understand this bit, if there is an inspect panel we remove viewpanel and edit panel? but you inspect panel from panel edit

Reply:
@torkel I may change this bit, I'm not happy with it but couldn't come up with anything better. 

Basically this makes sure that if you are in edit mode or view mode, the edit panel or view panel state is not turned off - because otherwise the state is updated to `undefined`. So those updates are no longer atomic as now it can be a combination of inspect panel + some other view at the same sine.